### PR TITLE
feat: updated checks to allow polygon documents

### PR DESCRIPTION
**Pull request by isaackps**
## Summary

Update document validation to allow polygon documents.

## Changes

- update processChainId fn to allow polygon docs

## Issues

linked to https://www.pivotaltracker.com/n/projects/2424361/stories/181365136, as once user created the config file they will create the documents and then verify the documents here.

### CODE REVIEW

File: src/utils/shared.ts [hunk 1]
```diff
@@ -54,16 +49,18 @@ export const getChainId = (rawDocument: WrappedOrSignedOpenAttestationDocument):
 
   const processChainId = (document: OpenAttestationDocument): number | undefined => {
     if (document.network) {
-      // Check for current bc, "ETH", and chainId, if need cater for other bc and network, update this accordingly. (V2)
-      if (document.network.chain === "ETH" && document.network.chainId) {
+      // Check for current blockchain, "ETH" or "MATIC", and chainId, if need cater for other blockchain and network, update this accordingly.
+      if (AvailableBlockChains.includes(document.network.chain) && document.network.chainId) {
```
Comment (simboonlong) [inline src/utils/shared.ts:53]: add test for `MATIC`? do check for a negative scenario too

File: src/constants/chain-info.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ export interface ChainInfoObject {
   };
 }
 
+export const AvailableBlockChains = ["ETH", "MATIC"];
+
```
Comment (simboonlong) [inline src/constants/chain-info.ts:17]: mmm so for polygon and polygon mumbai we need to use `MATIC` value eh? is this updated at document schema end?

Comment (simboonlong) [inline src/constants/chain-info.ts:17]: so for polygon and polygon mumbai we need to use `MATIC` chain value? just curious what happens when we use the wrong one

Comment (isaackps) [reply src/constants/chain-info.ts:17]: it will throw an error stating error in document, this check only happens for document that has the auto network switching. so if u have a polygon mumbai document as of now that doesn't have the auto networking switching, it will skip this check and proceed, given that the user selected the correct network from the dropdown.

File: src/utils/shared.ts (hunk 1 above)
Comment (isaackps) [reply src/utils/shared.ts:53]: ok

File: src/utils/shared.test.ts [hunk 3]
```diff
@@ -13,6 +13,22 @@ describe("getChainId for v2 document", () => {
     expect(getChainId(document)).toStrictEqual(5);
   });
 
+  it("should return the correct chainId for polygon mumbai network", () => {
+    const document = {
+      ...invoiceV2,
+      data: { ...invoiceV2.data, network: { chain: "MATIC", chainId: "80001" } },
+    } as unknown as WrappedOrSignedOpenAttestationDocument;
```
Comment (simboonlong) [inline src/utils/shared.test.ts:20]: the test shouldn't pass since MATIC is 137 no?

File: src/utils/shared.test.ts [hunk 4]
```diff
@@ -13,6 +13,22 @@ describe("getChainId for v2 document", () => {
     expect(getChainId(document)).toStrictEqual(5);
   });
 
+  it("should return the correct chainId for polygon mumbai network", () => {
```
Comment (simboonlong) [inline src/utils/shared.test.ts:16]: statement didn't tally with what was tested

File: src/utils/shared.test.ts (hunk 3 above)
Comment (isaackps) [reply src/utils/shared.test.ts:20]: nono, this matic is  the block chain, either ethereum: ETH or polygon: MATIC.

Comment (isaackps) [reply src/utils/shared.test.ts:20]: then the 80001 is testnet mumbai, 137 is polygon mainnet

File: src/utils/shared.test.ts (hunk 4 above)
Comment (isaackps) [reply src/utils/shared.test.ts:16]: how? the test is just reading the file that was provided and retrieve the chainId only arh.

**Review by simboonlong — APPROVED** [review]
